cpp/include/patterns/copy_v_transform_reduce_in_out_nbr.cuh

```diff
@@ -81,37 +81,40 @@ __global__ void for_all_major_for_all_nbr_low_degree(
   using weight_t      = typename GraphViewType::weight_type;
   using e_op_result_t = T;
 
-  auto const tid = threadIdx.x + blockIdx.x * blockDim.x;
-  auto idx =
-    static_cast<size_t>(row_first - matrix_partition.get_major_first()) + static_cast<size_t>(tid);
+  auto const tid          = threadIdx.x + blockIdx.x * blockDim.x;
+  auto major_start_offset = static_cast<size_t>(major_first - matrix_partition.get_major_first());
+  auto idx                = static_cast<size_t>(tid);
 
-  while (idx < static_cast<size_t>(row_last - matrix_partition.get_major_first())) {
+  while (idx < static_cast<size_t>(major_last - major_first)) {
     vertex_t const* indices{nullptr};
     weight_t const* weights{nullptr};
     edge_t local_degree{};
+    auto major_offset = major_start_offset + idx;
     thrust::tie(indices, weights, local_degree) =
-      matrix_partition.get_local_edges(static_cast<vertex_t>(idx));
+      matrix_partition.get_local_edges(static_cast<vertex_t>(major_offset));
 #if 1
     auto transform_op = [&matrix_partition,
                          &adj_matrix_row_value_input_first,
                          &adj_matrix_col_value_input_first,
                          &e_op,
-                         idx,
+                         major_offset,
                          indices,
                          weights] __device__(auto i) {
       auto minor        = indices[i];
       auto weight       = weights != nullptr ? weights[i] : weight_t{1.0};
       auto minor_offset = matrix_partition.get_minor_offset_from_minor_nocheck(minor);
       auto row          = GraphViewType::is_adj_matrix_transposed
                    ? minor
-                   : matrix_partition.get_major_from_major_offset_nocheck(idx);
+                   : matrix_partition.get_major_from_major_offset_nocheck(major_offset);
       auto col = GraphViewType::is_adj_matrix_transposed
-                   ? matrix_partition.get_major_from_major_offset_nocheck(idx)
+                   ? matrix_partition.get_major_from_major_offset_nocheck(major_offset)
                    : minor;
-      auto row_offset =
-        GraphViewType::is_adj_matrix_transposed ? minor_offset : static_cast<vertex_t>(idx);
-      auto col_offset =
-        GraphViewType::is_adj_matrix_transposed ? static_cast<vertex_t>(idx) : minor_offset;
+      auto row_offset = GraphViewType::is_adj_matrix_transposed
```

Comment:
I'd better avoid empty initializations, and also artificially creating a tuple when it really does not help much... Using a tuple can save one ```GraphViewType::is_adj_matrix_transposed``` (and note that this will be resolved in compile-time so this is really about code readability, not saving branches) but will require std::get<0>(tuple) and std::get<1>(tuple) instead of row_offset & col_offset.

I am not sure the gain (removing one GraphViewType::is_adj_matrix_transposed) is more than the loss (using std:::get<0>(tuple) and std::get<1>(tuple) and introducing a tuple type).

Let me know if I am mistaken.